agent/discovery/subscription.go

```diff
@@ -28,7 +30,11 @@ func (d SubscriptionDiscovery) GetId() string {
 func (d SubscriptionDiscovery) Discover() (string, error) {
 	subsData, err := subscription.NewSubscriptions()
 	if err != nil {
-		return "", err
+		if errors.Is(err, exec.ErrNotFound) {
+			return "No subscriptions were found. SUSEConnect tool is not available.", nil
```

Comment:
We should raise this with an error. If SUSEConnect is not available, the agent cannot run properly, so this is a critical state